fixham/pkg/tasks/hashrelease.go

```diff
@@ -0,0 +1,236 @@
+	}
+	manifestListName := fmt.Sprintf("%s/%s:%s", registry.QuayRegistry, operator.ImageName, operatorVersion)
+	if err = runner.ManifestPush(manifestListName, imageList, true); err != nil {
+		logrus.WithField("manifest list", manifestListName).WithError(err).Fatal("Failed to push operator manifest")
+	}
+	if err := runner.PushImage(fmt.Sprintf("%s/%s-init:%s", registry.QuayRegistry, operator.ImageName, operatorVersion)); err != nil {
+		logrus.WithError(err).Fatal("Failed to push operator init image")
+	}
+}
+
+// HashreleaseBuild builds the artificts hashrelease
+//
+// This includes the windows archive, helm archive, and manifests.
+func HashreleaseBuild(cfg *config.Config, outputDir string) {
+	hashreleaseOutputDir := hashreleaseDir(outputDir)
+	// TODO: ensure no changes in branch
+	if err := utils.CreateDir(hashreleaseOutputDir); err != nil {
+		logrus.WithError(err).Fatal("Failed to create hashrelease directory")
+	}
+	releaseVersion, err := hashrelease.RetrievePinnedCalicoVersion(outputDir)
+	if err != nil {
+		logrus.WithError(err).Fatal("Failed to get candidate name")
+	}
+	operatorVersion, err := hashrelease.RetrievePinnedOperatorVersion(outputDir)
+	if err != nil {
+		logrus.WithError(err).Fatal("Failed to get operator version")
+	}
+	if err := utils.ReleaseWindowsArchive(cfg.RepoRootDir, releaseVersion, hashreleaseOutputDir+"/files/windows"); err != nil {
+		logrus.WithError(err).Fatal("Failed to release windows archive")
+	}
+	if err := utils.HelmArchive(cfg.RepoRootDir, releaseVersion, operatorVersion, hashreleaseOutputDir); err != nil {
+		logrus.WithError(err).Fatal("Failed to release helm archive")
+	}
+	if err := utils.GenerateManifests(cfg.RepoRootDir, releaseVersion, operatorVersion, hashreleaseOutputDir); err != nil {
+		logrus.WithError(err).Fatal("Failed to generate manifests")
+	}
+
+	if err := command.Metadata(hashreleaseOutputDir, releaseVersion, operatorVersion); err != nil {
+		logrus.WithError(err).Fatal("Failed to generate metadata")
+	}
+}
+
+type result struct {
+	exists bool
+	err    error
+}
+
+func imgExists(imageName, imageRegistry string, ch chan result) {
+	r := result{}
+	exists, err := registry.ImageExists(imageName, imageRegistry)
+	r.exists = exists
+	r.err = err
+	ch <- r
+}
+
+// HashreleaseValidate validates the images in the hashrelease.
+// These images are checked to ensure they exist in the registry
+// as they should have been pushed in the standard build process.
+func HashreleaseValidate(cfg *config.Config, outputDir string) {
+	pinnedVersion, err := hashrelease.RetrievePinnedVersion(outputDir)
+	if err != nil {
+		logrus.WithError(err).Fatal("Failed to get pinned version")
+	}
+	images := pinnedVersion.Components
+	images["operator"] = pinnedVersion.TigeraOperator
+	results := make(map[string]result, len(images))
+
+	for name, component := range images {
+		ch := make(chan result)
+		go imgExists(component.Image+":"+component.Version, component.Registry, ch)
```

Comment:
I see a lot of `string + "" + string` throughout. 

Generally it's easier to read using `fmt.Sprintf` except for the most basic of uses.

e.g.,

```go
fmt.Sprintf("%s:%s", component.Image, component.Version)
```

In this case, you could also add another bound function to the component though, since it's attached to a struct already anyway.

e.g.,

```go
func (c Component) ImageWithTag() string {
  return fmt.Sprintf("%s:%s", c.Image, c.Tag)
}
```